lib/vtls/openssl.c

```diff
@@ -3822,7 +3822,11 @@ static size_t Curl_ossl_version(char *buffer, size_t size)
       sub[0]='\0';
   }
 
+#ifdef OPENSSL_FIPS
+  return msnprintf(buffer, size, "%s/%lx.%lx.%lx%s-fips",
+#else
   return msnprintf(buffer, size, "%s/%lx.%lx.%lx%s",
+#endif
```

Comment:
I would prefer to minimize the diff between the two conditions, so I'd prefer something like:

~~~c
   return msnprintf(buffer, size, "%s/%lx.%lx.%lx%s"
#ifdef OPENSSL_FIPS
                    "-fips"
#endif
                    ,
 ...

Reply:
I thought about this approach, but the `-fips` appended on the end of the string returned doesn't replace the `sub` -- so `OpenSSL/1.0.2k` still becomes `OpenSSL/1.0.2k-fips` and there's no string concatenation going on outside of what's going on in the `msnprintf` to just plop that `-fips` on there.

Having another `%s` in the format string to handle the `-fips` case still means we'd have a 5 line diff as we'd have to have an empty string for the for the `#else` condition and it doesn't seem much cleaner to me.

I can't think of a way _right now_ to keep it smaller.

Reply:
I don't understand. Surely my version of the code produces the same string output as your code does, both with and without the FIPS define set? Strings are concatenated automatically in C when written next to each other.

Reply:
I think I understand you now, I'll be making that change appropriately.

Reply:
so no dash now?